app/controllers/users_controller.rb

```diff
@@ -32,21 +32,23 @@ def new
   def create
     @user = User.new(user_params_with_ip)
 
-    if @user.spam?
-      @user.is_spam = true
-      # since we don't have any relations at this point yet,
-      # only perform soft-deletion
-      @user.soft_delete
-      flash[:error] = "Hrm, robots marked you as spam. If this was done in error, please email support@alonetone.com and magic fairies will fix it right up."
-      redirect_to logout_path
-    elsif is_a_bot? && @user.valid?
-      flash[:ok] = "We just sent you an email to '#{CGI.escapeHTML @user.email}'.<br/><br/>Just click the link in the email, and the hard work is over! <br/> Note: check your junk/spam inbox if you don't see a new email right away.".html_safe
-      redirect_to login_url(already_joined: true)
-    elsif @user.valid? && @user.save_without_session_maintenance
-      @user.reset_perishable_token!
-      UserNotification.signup(@user).deliver_now
-      flash[:ok] = "We just sent you an email to '#{CGI.escapeHTML @user.email}'.<br/><br/>Just click the link in the email, and the hard work is over! <br/> Note: check your junk/spam inbox if you don't see a new email right away.".html_safe
-      redirect_to login_url(already_joined: true)
+    if @user.valid?
+      if @user.spam?
+        @user.is_spam = true
+        @user.soft_delete
```

Comment:
Do we run into that same odd bug with `update_attributes` here, where the record will be saved?

Reply:
Well... Yes but I don't think it really matters since the user here is `valid?`
We could add this check after `bot?` and make `.save` explicit so at least it's obvious.
ThoughtS?

Reply:
My only concern with is `.save_without_session_maintenence` is how we prefer to save users on create — this ensures the user is not logged in on create (as they have to verify their email).

Reply:
But I haven't looked at that Authlogic stuff in a while, so.... maybe it's ok...

Reply:
But if we log out the user if the user is spam, doesn't it also ensures the user is not logged in?
If a user is spam, I don't think we need to even send an email to them. 
Thoughts? I'm going to find my user flow through #create action and paste it here.

Reply:
Stepping back a bit....We are in `User#create` — do we want to create a user record at all if we're spam?

Reply:
Right now, the code seems to rely on the bug being present (soft deleting a record that isn't yet created). Seems like if we want to go that route we should explicitly `save_without_session_maintenence` first to give us control over what's happening — if that bug is fixed, our code will still work.

Reply:
<img width="807" alt="Blank_Diagram__Lucidchart" src="https://user-images.githubusercontent.com/2320322/57744127-e246b480-7695-11e9-91de-74e3019b0c2c.png">

_not at all overkill_

Reply:
I think we should create a spam user... what if user is not spam afterall? 
I pushed the `.save_without_session_maintenence` already... so that should answer this question? 

I was wondering if we can assign deleted_at time to user in the helper (rather than calling upadate_attributes), but it just doesn't seem... both necessary and "safe" :) How many times are you really going to be dealing with `User.new` object in your app...